api/src/database/run-ast.ts

```diff
@@ -334,7 +332,16 @@ async function getDBQuery(
 					orderByString,
 					orderByFields,
 				);
+
+				// Only order by directus_row_number, as the directus_row_number is derived from a window function that is
+				// ordered by the sort fields within every primary key partition. Since the only relevant result is the first
+				// row of this partition, the order by here ensures that all rows with a directus_row_number = 1 show up first
+				// in the inner query result, and are correctly truncated by the limit, but not earlier.
+				orderByString = `?? asc, ${orderByString}`;
+				orderByFields.unshift(knex.ref('directus_row_number'));
```

Comment:
@hanneskuettner Love this fantastic solution to sort by the row number. I've patched it by including the original query sorting. For your review please~ 🫡

Reply:
Very good find! As discussed privately, adding the original query sorting, in addition to the row number, will ensure that the the results are correctly sorted in the inner query, if there are ever more results than the query limit, which was overlooked by me 👍